source/isaaclab/isaaclab/envs/mdp/actions/pink_task_space_actions.py

```diff
@@ -56,7 +58,11 @@ def __init__(self, cfg: pink_actions_cfg.PinkInverseKinematicsActionCfg, env: Ma
         assert env.num_envs > 0, "Number of environments specified are less than 1."
         self._ik_controllers = []
         for _ in range(env.num_envs):
-            self._ik_controllers.append(PinkIKController(cfg=copy.deepcopy(self.cfg.controller), device=self.device))
+            self._ik_controllers.append(
+                PinkIKController(
+                    cfg=copy.deepcopy(self.cfg.controller), robot_cfg=env.scene.cfg.robot, device=self.device
```

Comment:
Deepcopy doesn't work all the time with nested configclass instances. Using `.copy()` should be safer.

```suggestion
                    cfg=self.cfg.controller.copy(), robot_cfg=env.scene.cfg.robot, device=self.device
```

Reply:
done